src/Symfony/Bridge/Monolog/Handler/ConsoleHandler.php

```diff
@@ -64,6 +65,7 @@ public function __construct(
         bool $bubble = true,
         array $verbosityLevelMap = [],
         private array $consoleFormatterOptions = [],
+        private ?InputInterface $input = null,
```

Comment:
Putting the input in the constructor won't play well, because the input is not a service in the DI controller. It should be handled the same way than the output (and injected in the `onCommand` event listener method)

Reply:
i also expected a new flag `interactive_only` https://github.com/symfony/symfony/issues/58715#issuecomment-2448015549

Reply:
@ro0NL there's no flag here since the config is in the Monolog bundle, here it reacts to the input being set, which will get set (or not) based on the flag in the config, it's a separate repo.

Reply:
@stof not sure what's the difference between output and input, since output is injected via ctor? :thinking: Or is that not used, I can also see `setOutput()` for some reason, why are they both there?

Reply:
@ro0NL now I understood what you've meant, basically we can't rely on the input being there or not because it will always get injected (and not, as I expected, be created by a factory which contains the config tree), so we must have a flag here too.

Reply:
in the actual usage in the bundle, the output is **not** injected in the constructor (because it **cannot** inject it there as the output is not managed by the container). The constructor argument is usable for custom usages of the bridge.

In the bundle, the current output is managed by the `onCommand` method listening to the event triggered by the console application.

and this input will be injected independently of the configuration, so you still need to inject the boolean setting.

Reply:
@stof :point_up:

Reply:
I suggest providing only the setter for the input, not the constructor argument. There is no reason to make the constructor signature more complex for something we won't be able to use anyway (the output is in the constructor for legacy reasons, while not being worth doing a deprecation)

Reply:
I suppose we can now remove the constructor argument.

Reply:
@stof @fabpot done.